airflow/sensors/external_task_sensor.py

```diff
@@ -68,23 +70,35 @@ def __init__(self,
                  external_dag_id,
                  external_task_id=None,
                  allowed_states=None,
+                 failed_states=None,
                  execution_delta=None,
                  execution_date_fn=None,
                  check_existence=False,
                  *args,
                  **kwargs):
         super().__init__(*args, **kwargs)
         self.allowed_states = allowed_states or [State.SUCCESS]
+        self.failed_states = failed_states or []
+
+        total_states = []
+        total_states.extend(self.allowed_states)
+        total_states.extend(self.failed_states)
+
+        if len(list(set(total_states))) < (len(self.failed_states) + len(self.allowed_states)):
```

Comment:
```suggestion
        if set(self.failed_states).intersection(set(self.allowed_states)):
```
WDYT? Personally I would cast states to set even earlier.